queries.sql

```diff
@@ -83,4 +83,63 @@ WHERE owners.full_name = 'Dean Winchester' AND animals.escape_attempts = 0;
 
 SELECT owners.full_name, count(animals.id)
 FROM owners LEFT JOIN animals ON owners.id = animals.owners_id
-GROUP BY owners.full_name;
\ No newline at end of file
+GROUP BY owners.full_name;
+
+SELECT a.name  FROM visits as v 
+JOIN animals AS a ON v.animal_id = a.id 
+JOIN vets AS vt ON v.vet_id = vt.id 
+WHERE vt.name = 'William Tatcher' 
+ORDER BY v.date DESC LIMIT 1;
+
+SELECT COUNT(DISTINCT v.animal_id) AS total_animals_seen 
+FROM visits v 
+JOIN vets vt ON v.vet_id = vt.id 
+WHERE vt.name = 'Stephanie Mendez';
+
+SELECT v.name AS vet_name, COALESCE(specialty, 'No Specialty') AS specialty
+FROM vets v
+LEFT JOIN (SELECT vet_id, STRING_AGG(species.name, ', ') AS specialty
+FROM specializations
+JOIN species ON specializations.species_id = species.id 
+GROUP BY vet_id) s ON v.id = s.vet_id 
+ORDER BY name, specialty;
+
+SELECT a.name AS animal_name, v.date 
+FROM visits v 
+JOIN animals a ON v.animal_id = a.id 
+WHERE v.vet_id = 3  AND v.date BETWEEN '2020-04-01' AND '2020-08-30';
+
+SELECT a.name AS animal_name, COUNT(*) AS visit_count 
+FROM visits v 
+JOIN animals a ON v.animal_id = a.id 
+GROUP BY a.name 
+ORDER BY visit_count DESC LIMIT 1;
+
+SELECT vet.name AS vet_name, MIN(v.date) AS first_visit_date 
+FROM visits v 
+JOIN vets vet ON v.vet_id = vet.id 
+JOIN animals a ON v.animal_id = a.id 
+WHERE vet.name = 'Maisy Smith' 
+GROUP BY vet.name;
```

Comment:
- [ ] In this query you are returning the vet name however according to the requirements you're supposed to return the animal name who visited Maisy last